src/std/make.ss

```diff
@@ -592,6 +597,9 @@ TODO:
   (match spec
     ((? string? modf)
      (gxc-compile modf #f settings #t))
+    ([gxc: modf [submodules: submodules . _] . opts]
+     (for-each (cut build <> settings) submodules)
```

Comment:
The for-each sounds very wrong and contrary to the principles of make: make a queue of things to be scheduled according to dependency order, rather than execute immediately out of a plan (which can lead to out-of-dependency-order execution and/or sometimes double execution).

If things are not currently done, they need to be queued as dependencies, not done immediately.

Reply:
Sorry, but the code right now is completely broken; submodules never get built.
If you can suggest a better fix, fine, but iam inclined to keep it as is (and i dont fully understand the code the way it has morphed)

Reply:
OK, after conversation with vyzo, it looks like the ssi entry also uses for-each build, and that I had just forgotten the same thing for gxc: after patiently building it in normalize-buildspec, so it doesn't look that wrong after all.

This part of the code is very much out-of-cache for me right now, and obviously under-documented. But the fix now looks right-ish. I will have to make another pass at documenting things whenever I put things back into cache.